CIP-0164/README.md

```diff
@@ -0,0 +1,2797 @@
+<a name="figure-19" id="figure-19"></a>
+
+![Leios Variants Comparison](images/leios-variants-comparison-radar.svg)
+
+<em>Figure 19: Comparison: Praos (red), proposed Leios (teal), and research
+Leios (orange)</em>
+
+</div>
+
+The comparison illustrates the fundamental trade-off between throughput capacity
+and ecosystem disruption. While the research paper's approach (orange) achieves
+higher throughput, it requires extensive ecosystem changes, significantly longer
+confirmation times (2-3 minutes), and extended development timelines (2.5-3
+years). The proposed specification (teal) strikes a strategic balance,
+delivering substantial throughput improvements while maintaining manageable
+latency increases and ecosystem compatibility.
+
+**Key Trade-offs:**
+
+- **Throughput**: 30-65x increase (from ~4.5 TxkB/s to ~140-300 TxkB/s)
+  addresses economic sustainability and Reserve depletion timeline
+- **Transaction Inclusion Latency**: Increases from ~20 seconds to 45-60 seconds
+- **Infrastructure**: Requires modest hardware upgrades (4-core machines) and
+  cryptographic key registration, offset by enhanced rewards
+- **Time to Market**: ~1-1.5 years for deployment versus 2.5-3 years for
+  higher-concurrency alternatives
+
+**Protocol Limitations:**
+
+The specification involves operational constraints including timing dependencies
+that require careful parameter tuning, confirmation complexity through multiple
+transaction states, and interdependent protocol parameters requiring ongoing
+optimization. These represent acceptable trade-offs for achieving substantial
+throughput improvements while maintaining familiar transaction structures and
+ecosystem compatibility.
+
+**Security Considerations:**
+
+The proposed specification maintains Praos security properties through formal
+timing constraints that ensure RB processing stays within Praos bounds. The
+high-participation design (75% voting threshold) eliminates invalid transactions
+in RBs and provides strong network assumptions for certified EB propagation. New
+threats include equivocation by EB producers/voters and transaction availability
+attacks, mitigated through cryptographic validation, equivocation detection, and
+the high voting threshold requirement. Comprehensive analysis is documented in
+the [Protocol Security](#protocol-security) section and
+[threat model](https://github.com/input-output-hk/ouroboros-leios/blob/d5f1a9bc940e69f406c3e25c0d7d9aa58cf701f8/docs/threat-model.md).
+
+### Design Decisions
+
+#### Transaction References in Endorser Blocks
+
+EBs contain transaction references rather than full transaction bodies to avoid
+re-transmitting data already diffused through the network. Nodes fetch only
+transactions they don't possess, reducing bandwidth and avoiding redundant
+validation when transactions are endorsed by multiple EBs. Complete transaction
+data remains available through network diffusion and local storage for immutable
+chain purposes.
```

Comment:
This seems to bury the lede that EB TXs must be stored out-of-band from RBs and EBs, since the TX body will not be contained in either.